independent-projects/tools/devtools-common/src/main/java/io/quarkus/platform/tools/ToolsUtils.java

```diff
@@ -112,7 +113,11 @@ public static ExtensionCatalog resolvePlatformDescriptorDirectly(String bomGroup
                 }
             }
             if (platformJson == null) {
-                throw new RuntimeException("Failed to resolve the default platform JSON descriptor", e);
+                throw new RuntimeException("Failed to resolve extension catalog for " +
```

Comment:
The error message is better but for the case of the mvn example (which is also possible with quarkus cli) the root cause is not that the registry catalog cannot be found but that the specified platform does not resolve at all, right?

Reply:
i.e. wouldn't a better message be something like:

"The requested platform com.redhat.quarkus:quarkus-bom:2.2.3.SP2-redhat-00001 not found." 

The missing json descriptor or extension catalog is secondary.

Reply:
The thing is that we are not even resolving BOMs here, we are resolving extension catalogs (the JSON) and this is what fails in this case. It could fail for various reasons: misspelled group/artifact/version, missing or incorrect Maven repo config.

Reply:
"Missing" or "not found" could be interpreted as it was looked up in proper places and could not be found, which wouldn't be the case if proper Maven repos weren't configured or aren't available for some reason. A "resolution failure" is more general and, imo, more appropriate message here. I added some hints for the user what to look for in this case, how is this looking?
````
[INFO] --- quarkus-maven-plugin:999-SNAPSHOT:create (default-cli) @ standalone-pom ---
[WARNING] Failed to resolve extension catalog of com.redhat.quarkus:quarkus-bom:pom:2.2.3.SP2-redhat-00001 from Maven repository central (https://repo.maven.apache.org/maven2)
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  2.395 s
[INFO] Finished at: 2022-02-07T14:23:27+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:create (default-cli) on project standalone-pom: Execution default-cli of goal io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:create failed: Failed to resolve extension catalog for com.redhat.quarkus:quarkus-bom:pom:2.2.3.SP2-redhat-00001. Make sure the groupId, artifactId and version are spelled corrently and the relevant Maven repositories are configured.: Failed to resolve artifact com.redhat.quarkus:quarkus-bom-quarkus-platform-descriptor:json:2.2.3.SP2-redhat-00001:2.2.3.SP2-redhat-00001: com.redhat.quarkus:quarkus-bom-quarkus-platform-descriptor:json:2.2.3.SP2-redhat-00001:2.2.3.SP2-redhat-00001 was not found in https://repo.maven.apache.org/maven2 during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of central has elapsed or updates are forced -> [Help 1]
````
````
[aloubyansky@localhost playground]$ qs create -P com.redhat.quarkus:quarkus-bom:2.2.3.SP2-redhat-00001
Creating an app (default project type, see --help).
[ERROR] ❗  Unable to create project: Failed to resolve extension catalog for com.redhat.quarkus:quarkus-bom:pom:2.2.3.SP2-redhat-00001. Make sure the groupId, artifactId and version are spelled correctly and the relevant Maven repositories are configured.
````